spec/models/stats_spec.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe Stats do
```

Comment:
One of the things you can do in RSpec is explicitly define the subject in your tests, and then reuse that in them.

Using the first test as an example, we could tweak it to do this

```ruby
RSpec.describe Stats, type: :model do
  subject { described_class.new }

  describe "#new_reg" do
    it "returns the correct value for new_reg" do
      expect(subject.new_reg).to eq(25)
    end
  end
end
```

This saves having to repeat the setup line of `stats = Stats.new` in each of the tests 😁 

It also has the handy `described_class` which means should you rename your class, you only need to change the top level line of the test.

Finally we normally are explicit in defining the `:type` in our rspec tests. Rails and rspec can normally infer this from the directory its placed in. But because we deal a lot with engines and funny types like forms and services, we've chosen to just be explicit to make it clear.

Reply:
Thanks Alan, sounds good! I've amended the tests in line with all this. I also changed "returns the correct value for new_reg" etc. to just "returns the correct value" since it seemed redundant to have an error like "Stats#new_reg returns the correct value of new_reg" !

Reply:
Awesome idea! It's not required, but my approach when writing tests is to always run them with these additional flags `bundle exec rspec --format documentation`. This prints out the `desc`, `context` and `it` titles, and means I can read them and confirm they make sense as a sentence

```bash
Stats
  #email_renewals
    returns the correct value for email_renewals
  #assisted_pc
    returns the correct value for assisted_pc
  #new_reg
    returns the correct value for new_reg
```